cmd/utils/flags.go

```diff
@@ -387,6 +387,12 @@ var (
 		Value:    ethconfig.Defaults.TransactionHistory,
 		Category: flags.StateCategory,
 	}
+	BlockHistoryFlag = &cli.Uint64Flag{
+		Name:     "history.blocks",
+		Usage:    "Number of recent blocks to maintain in DB (default = 0, 0 = entire chain). Notice!! it only prunes old blocks when starting, and pruning is not involving TxIndex/bloomIndex.",
```

Comment:
Wouldn't this work for the entire node cycle? It says it only deletes blocks once at startup?

Reply:
@galaio

Reply:
accepted